shared/subspace-data-retrieval/Cargo.toml

```diff
@@ -15,16 +15,20 @@ include = [
 anyhow.workspace = true
 async-trait.workspace = true
 futures.workspace = true
-hex.workspace = true
+hex = { workspace = true, features = ["std"] }
 parity-scale-codec = { workspace = true, features = ["derive"] }
 subspace-archiving.workspace = true
-subspace-core-primitives.workspace = true
+subspace-core-primitives = { workspace = true, features = ["std"] }
 subspace-erasure-coding.workspace = true
+subspace-runtime-primitives = { workspace = true, features = ["std"] }
```

Comment:
This line is a problem. It pulls a huge tree of Substrate crates into farmer components and other places just to get access to a single constant. This is extremely unfortunate and slows compilation time substantially.

Data retrieval code was supposed to be generic and not depend on Substrate in any way.

Reply:
Ok, I'll fix that in #3472 tomorrow.

Reply:
This is now fixed in 1a9ccacacdde58d9ada6eff76720404ed8dc2855 in #3472, and I left a comment so it doesn't happen again. (I think this is the second or third time.)